crates/wasm-smith/src/lib.rs

```diff
@@ -104,9 +104,6 @@ pub struct Module {
     config: Rc<dyn Config>,
     valtypes: Vec<ValType>,
 
-    /// Outer modules, if any (used for module linking)
-    outers: Vec<Outer>,
```

Comment:
We talked a bit at the wasmtime meeting yesterday, but to copy over some of what we concluded I believe the current thinking is that we may want to hold off on the deletion of the current module linking support until there's a better sense of the timeline of what it takes to replace it with the current component model writeup.

I don't think that means that you should implement module-linking for the new component model here-and-now but I do think that it may delay landing this PR. That being said I also think that all this support in `wasm-smith` would be somewhat useful to not throw away since I'm at least naively under the impression that it should be largely reusable for the next iteration of wasm-smith-module-linking-using-modules. The main difference between the historical module linking implementation and the current one is what types of items can be in a module and the representation of types, which feels like it should be a smaller delta from the current `wasm-smith` implementation than deleting everything and bringing it all back.

In any case I think the basic tl;dr; is that we may want to figure out a timeline for implementing all the various components since there was some pushback to having module linking support removed for now.

Reply:
That's fine, there's no rush to get this merged in, the component tooling being built on top of these changes can work off my branch for now.